appcues/src/main/java/com/appcues/ui/ExperienceRenderer.kt

```diff
@@ -169,24 +165,22 @@ internal class ExperienceRenderer(
 
     suspend fun start(owner: StateMachineOwning, context: RenderContext) {
         // If there's already a frame for the context, reset it back to its unregistered state.
-        val existingOwner = stateMachines.getOwner(context) as? AppcuesFrameStateMachineOwner
-        if (existingOwner != null) {
-            existingOwner.frame.get()?.reset()
-            existingOwner.stateMachine = null
-        }
+        stateMachines.getOwner(context)?.reset()
 
         // If the machine being started is already registered for a different context,
         // reset it back to its unregistered state before potentially showing new content.
-        if (owner.stateMachine != null) {
-            (owner as? AppcuesFrameStateMachineOwner)?.let {
-                it.frame.get()?.reset()
-                it.stateMachine = null
-            }
-        }
+        owner.reset()
+
+        // if the owner already has a state machine, make sure it's lifecycle tracking is stopped before
+        // we replace it with a new one
+        owner.stateMachine?.stopLifecycleTracking()
```

Comment:
@andretortolano the second commit here now handles this - and this line in particular is where we stop any existing listeners when re-creating a state machine for a new frame registration.

`StateMachine` now owns its own `ExperienceLifecycleTracker`, and starts it on init. The tracker is a `factory` in Koin now, with each `StateMachine` getting its own instance. The tracker keeps track of the `Job` references for its observers, and they can be canceled when it is stopped.